vogue/server/utils/bioinfo/mip.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python
+
+from vogue.constants.constants import MONTHS, MIP_DNA_PICARD
+
+
+def mip_dna_picard_time_plot(adapter, year: int) -> dict:
+    """Prepares data for the MIP picard over time plot."""
+
+    pipe = [{
+        '$lookup': {
+            'from': 'sample',
+            'localField': '_id',
+            'foreignField': '_id',
+            'as': 'sample_info'
+        }
+    }, {
+        '$unwind': {
+            'path': '$sample_info'
+        }
+    }, {
+        '$match': {
+            'sample_info.received_date': {
+                '$exists': 'True'
+            }
+        }
+    }, {
+        '$project': {
+            'mip-dna': 1,
+            'received_date': '$sample_info.received_date'
+        }
+    }, {
+        '$match': {
+            'received_date': {
+                '$exists': 'True'
+            }
+        }
+    }, {
+        '$project': {
+            'month': {
+                '$month': '$received_date'
+            },
+            'year': {
+                '$year': '$received_date'
+            },
+            'mip-dna': 1
+        }
+    }, {
+        '$match': {
+            'year': {
+                '$eq': int(year)
+            }
+        }
+    }]
+    aggregate_result = adapter.bioinfo_samples_aggregate(pipe)
+    final_data = {}
+    for data in MIP_DNA_PICARD.values():
+        for key in data:
+            final_data[key] = []
+
+    for sample in aggregate_result:
+        sample_id = sample['_id']
+        mip_dna_analysis = sample.get('mip-dna')
+        if mip_dna_analysis:
+            multiqc_picard_insertSize = mip_dna_analysis.get(
```

Comment:
Flatten with
```suggestion
        if not mip_dna_analysis:
            continue
        multiqc_picard_insertSize = mip_dna_analysis.get(
```
then everything can be moved one block to the left

Reply:
Also consider reactor the loop or what is in the loop into it's own functions